File: python/bizx/crypto_wallet.py
from dataclasses import dataclass
from enum import Enum

class Coin(str, Enum):
    BTC='BTC'; BCH='BCH'; LTC='LTC'; DOGE='DOGE'; ETH='ETH'; ETC='ETC'; SOL='SOL'; ADA='ADA'; XRP='XRP'; DOT='DOT'; AVAX='AVAX'

@dataclass(frozen=True)
class Endpoint:
    host: str
    port: int
    transport: str = 'HTTPS'

@dataclass
class Wallet:
    coin: Coin
    address: str
    watch_only: bool = True

    def transaction(self, to: str, amount: str) -> dict:
        return {'state':'awaiting_confirmation','coin':self.coin.value,'from':self.address,'to':to,'amount':amount}

class Network:
    @staticmethod
    def endpoint(host: str, port: int, transport: str='HTTPS') -> Endpoint:
        if transport not in {'TCP','UDP','HTTP','HTTPS','WebSocket'}: raise ValueError('unsupported transport')
        return Endpoint(host, port, transport)

File: ruby/lib/bizx/crypto_wallet.rb
module BizX
  class CryptoWallet
    COINS = %w[BTC BCH LTC DOGE ETH ETC SOL ADA XRP DOT AVAX].freeze
    attr_reader :coin, :address, :watch_only
    def initialize(coin, address, watch_only: true)
      raise ArgumentError, 'unsupported coin' unless COINS.include?(coin)
      @coin, @address, @watch_only = coin, address, watch_only
    end
    def transaction(to, amount)
      { state: 'awaiting_confirmation', coin: @coin, from: @address, to: to, amount: amount }
    end
  end
end
